helix-core/src/main/java/org/apache/helix/messaging/handling/HelixStateTransitionHandler.java

```diff
@@ -264,8 +271,12 @@ void postHandleMessage() {
 
     try {
       // Update the ZK current state of the node
-      PropertyKey key = keyBuilder.currentState(instanceName, sessionId, resource,
-          bucketizer.getBucketName(partitionKey));
+      PropertyKey key =
+          _isTaskMessage && !Boolean.getBoolean(SystemPropertyKeys.TASK_CURRENT_STATE_PATH_DISABLED)
```

Comment:
Let's have a variable to track TASK_CURRENT_STATE_PATH_DISABLED. We dont have to do the parse every time. Also, let's give a default value as true since participant will be the last component to deploy.

Reply:
Default value should be false because it's "disabled" right? If participants get the new version of Helix, we should assume that they want to use the new path, so "disabled" should default to false. 

For variable tracking, there is a problem: if the boolean value is recorded at a central place, users cannot update this system property during run time, right?  If there is a variable that's tracking it, then it's as if it's cached.

Reply:
Ah... Yeah, the "disable" should be false.

But for dynamic update, let's not overkill the design:
1. For Helix, we use ZK centralized config for dynamic update purpose. If we decide to use system property, it means it is not dynamic changeable.
2. The system property mostly comes from Java property, which require restart the application. So we cannot support dynamic update as well.

Make it a variable can also simplify the code. I would suggest create a variable to hold that when we create the object.

Reply:
Discussion ongoing.

Reply:
Concluded to use one local variable when necessary.